### PR TITLE
Mock permissions depending on username

### LINKED ISSUE #142: Permissions in testing mode
When testing code that uses pymesync, it'd be very helpful to be able to `get_users()` a user that has admin permissions, etc.

### PR DESCRIPTION
**Pull request by subnomo**
Fixes #142 

## Changes in this PR.

- [X] `get_users` returns true for permissions matching username
- [x] Update documentation with example
- [x] Add tests

## Testing this PR.

1. Run `make verify`
2. See tests and flake pass
3. Read the additions to docs/source/testing.rst

@osuosl/devs

### CODE REVIEW

File: docs/source/testing.rst [hunk 1]
```diff
@@ -148,6 +148,15 @@ An (almost) exhaustive example of test mode:
...
   >>>
+  >>> ts.get_users("admin")
+  [{'username': 'admin', 'display_name': 'X. Ample User', 'site_manager': False, 'site_admin': True, 'site_spectator': False, 'created_at': '2015-02-29', 'active': True, 'deleted_at': None, 'email': 'example@example.com'}]
+  >>>
+  >>> ts.get_users("manager")
+  [{'username': 'admin', 'display_name': 'X. Ample User', 'site_manager': True, 'site_admin': False, 'site_spectator': False, 'created_at': '2015-02-29', 'active': True, 'deleted_at': None, 'email': 'example@example.com'}]
```
Comment (matthewrsj) [inline docs/source/testing.rst]: Should this be `'username': 'manager'`?

**Comment by ElijahCaine** [comment]
This seems to do the thing it's supposed to do. +1

File: docs/source/testing.rst (hunk 1 above)
Comment (subnomo) [inline docs/source/testing.rst]: Yes, thank you.

**Comment by aaroncohen73** [comment]
+1

**Comment by ElijahCaine** [comment]
@subnomo feel free to close this if you like it.

**Comment by Kennric** [comment]
+1

**Pull request merged**